packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -140,3 +203,74 @@ export function unstable_httpSubscriptionLink<
     };
   };
 }
+
+/**
+ * We wrap EventSource so that is can be reinitialized with new options
+ */
+class EventSourceWrapper implements Partial<EventSource> {
+  private es: EventSource;
+
+  private listeners: Partial<Record<keyof EventSourceEventMap, any[][]>> = {};
+
+  constructor(url: string, options: EventSourceInit | undefined) {
+    this.es = new EventSource(url, options);
+  }
+
+  restart(url: string, options: EventSourceInit | undefined) {
+    for (const type in this.listeners) {
+      for (const [t, l, o] of this.listeners[
+        type as keyof EventSourceEventMap
+      ] ?? []) {
+        this.es.removeEventListener(t as string, l, o);
+      }
+    }
+    this.es.close();
+    this.es = new EventSource(url, options);
+    for (const type in this.listeners) {
+      for (const [t, l, o] of this.listeners[
+        type as keyof EventSourceEventMap
+      ] ?? []) {
+        this.es.addEventListener(t as string, l, o);
+      }
+    }
+  }
+
+  close() {
+    this.listeners = {};
+    this.es.close();
+  }
+
+  getEventSource() {
+    return this.es;
+  }
+
+  get readyState() {
+    return this.es.readyState;
+  }
+
+  addEventListener<TEvent extends keyof EventSourceEventMap>(
+    type: TEvent,
+    listener: (this: EventSource, ev: EventSourceEventMap[TEvent]) => any,
+    options?: boolean | AddEventListenerOptions,
+  ) {
+    this.listeners[type] ??= [];
+    this.listeners[type].push([type, listener, options]);
+
+    this.es.addEventListener(type, listener, options);
+  }
+
+  removeEventListener<TEvent extends keyof EventSourceEventMap>(
+    type: TEvent,
+    listener: (this: EventSource, ev: EventSourceEventMap[TEvent]) => any,
+    options?: boolean | EventListenerOptions,
+  ) {
+    this.listeners[type] ??= [];
+    for (const [t, l, o] of this.listeners[type] ?? []) {
+      if (t === type && l === listener) {
+        this.listeners[type].splice(this.listeners[type].indexOf([t, l, o]), 1);
+      }
+    }
```

Comment:
this can be simplified to something like

```ts
const idx = this.listeners.findIndex(x => /* ... */)
const listener = this.listeners[idx]
if (!listener) {
  return
}
this.es.removeEventListener(listener.type, listener.callback, listener.options)

// ... splice
```